oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/name/ReadWriteNamespaceRegistry.java

```diff
@@ -71,11 +71,13 @@ public void registerNamespace(String prefix, String uri)
 
         // sanity check for legal namespace names (excluding the "internal"
         // namespace, see OAK-74)
-        if (!NamespaceConstants.NAMESPACE_REP.equals(uri)) {
-            if (!uri.contains(":")) {
+        if (!NamespaceConstants.NAMESPACE_REP.equals(uri) && !uri.contains(":")) {
+            if (Boolean.parseBoolean(System.getProperty("oak.allowInvalidNamespaceUris", "true"))) {
```

Comment:
I'd suggest using a SystemPropertySupplier.

Reply:
Done in https://github.com/apache/jackrabbit-oak/pull/2258/commits/e90fc3bf65d965728c8f7a74183ab8098ffe8acb

Reply:
Making it static is conflicting with testing this feature flag...